### ISSUE TITLE
Flow Loss is increasing after multiple iterations

### ISSUE THREAD

**Issue by YoushaaMurhij** [HugoCasa/STrajNet-torch#1]
Hi, thanks for sharing your code!
I tried to train it on WOD, but the flow loss is increasing!
Could you please help me figure out what could be the problem?
Thanks!

```
obs. loss: 0.243648, occl. loss:  0.150102, flow loss: 44.460011, warp loss: 1.001977  [ 3096/    0]
obs. loss: 0.243114, occl. loss:  0.149444, flow loss: 44.516956, warp loss: 1.008347  [ 3120/    0]
obs. loss: 0.242311, occl. loss:  0.148761, flow loss: 44.644539, warp loss: 1.012815  [ 3144/    0]
obs. loss: 0.241913, occl. loss:  0.148108, flow loss: 44.702065, warp loss: 1.020886  [ 3168/    0]
obs. loss: 0.241336, occl. loss:  0.147460, flow loss: 44.820835, warp loss: 1.027358  [ 3192/    0]
obs. loss: 0.240734, occl. loss:  0.146894, flow loss: 45.005154, warp loss: 1.032728  [ 3216/    0]
obs. loss: 0.240699, occl. loss:  0.146575, flow loss: 45.212120, warp loss: 1.044246  [ 3240/    0]
obs. loss: 0.240347, occl. loss:  0.146173, flow loss: 45.515354, warp loss: 1.051889  [ 3264/    0]
obs. loss: 0.240065, occl. loss:  0.145916, flow loss: 45.925617, warp loss: 1.060386  [ 3288/    0]
obs. loss: 0.239506, occl. loss:  0.145567, flow loss: 46.405163, warp loss: 1.064121  [ 3312/    0]
obs. loss: 0.239291, occl. loss:  0.145209, flow loss: 47.343037, warp loss: 1.072092  [ 3336/    0]
obs. loss: 0.239230, occl. loss:  0.144918, flow loss: 48.340939, warp loss: 1.082763  [ 3360/    0]
obs. loss: 0.239001, occl. loss:  0.144661, flow loss: 51.219795, warp loss: 1.086341  [ 3384/    0]
obs. loss: 0.238759, occl. loss:  0.144391, flow loss: 54.465942, warp loss: 1.090015  [ 3408/    0]
obs. loss: 0.238499, occl. loss:  0.144047, flow loss: 57.888397, warp loss: 1.093878  [ 3432/    0]
obs. loss: 0.238189, occl. loss:  0.143617, flow loss: 61.946064, warp loss: 1.096697  [ 3456/    0]
obs. loss: 0.237959, occl. loss:  0.143296, flow loss: 67.472435, warp loss: 1.099877  [ 3480/    0]
obs. loss: 0.237651, occl. loss:  0.142785, flow loss: 76.707634, warp loss: 1.101141  [ 3504/    0]
obs. loss: 0.237271, occl. loss:  0.142269, flow loss: 86.882683, warp loss: 1.101937  [ 3528/    0]
obs. loss: 0.236673, occl. loss:  0.141731, flow loss: 97.023819, warp loss: 1.100915  [ 3552/    0]
```

**Comment by YoushaaMurhij** [comment]
Here are some visual logs.  I think '_sigmoid_xe_warp_loss' could be leading to this issue.
https://api.wandb.ai/links/youshaamurhij/oixlz0st

**Comment by hugocasa** [comment]
Hi, thanks for raising the issue and the logs. 

We also noticed this, i'll try to fix it this week.

**Comment by flclain** [comment]
Found a similar phenomenon, did you solve this problem?

**Comment by flclain** [comment]
> Hi, thanks for sharing your code! I tried to train it on WOD, but the flow loss is increasing! Could you please help me figure out what could be the problem? Thanks!
> 
> ```
> obs. loss: 0.243648, occl. loss:  0.150102, flow loss: 44.460011, warp loss: 1.001977  [ 3096/    0]
> obs. loss: 0.243114, occl. loss:  0.149444, flow loss: 44.516956, warp loss: 1.008347  [ 3120/    0]
> obs. loss: 0.242311, occl. loss:  0.148761, flow loss: 44.644539, warp loss: 1.012815  [ 3144/    0]
> obs. loss: 0.241913, occl. loss:  0.148108, flow loss: 44.702065, warp loss: 1.020886  [ 3168/    0]
> obs. loss: 0.241336, occl. loss:  0.147460, flow loss: 44.820835, warp loss: 1.027358  [ 3192/    0]
> obs. loss: 0.240734, occl. loss:  0.146894, flow loss: 45.005154, warp loss: 1.032728  [ 3216/    0]
> obs. loss: 0.240699, occl. loss:  0.146575, flow loss: 45.212120, warp loss: 1.044246  [ 3240/    0]
> obs. loss: 0.240347, occl. loss:  0.146173, flow loss: 45.515354, warp loss: 1.051889  [ 3264/    0]
> obs. loss: 0.240065, occl. loss:  0.145916, flow loss: 45.925617, warp loss: 1.060386  [ 3288/    0]
> obs. loss: 0.239506, occl. loss:  0.145567, flow loss: 46.405163, warp loss: 1.064121  [ 3312/    0]
> obs. loss: 0.239291, occl. loss:  0.145209, flow loss: 47.343037, warp loss: 1.072092  [ 3336/    0]
> obs. loss: 0.239230, occl. loss:  0.144918, flow loss: 48.340939, warp loss: 1.082763  [ 3360/    0]
> obs. loss: 0.239001, occl. loss:  0.144661, flow loss: 51.219795, warp loss: 1.086341  [ 3384/    0]
> obs. loss: 0.238759, occl. loss:  0.144391, flow loss: 54.465942, warp loss: 1.090015  [ 3408/    0]
> obs. loss: 0.238499, occl. loss:  0.144047, flow loss: 57.888397, warp loss: 1.093878  [ 3432/    0]
> obs. loss: 0.238189, occl. loss:  0.143617, flow loss: 61.946064, warp loss: 1.096697  [ 3456/    0]
> obs. loss: 0.237959, occl. loss:  0.143296, flow loss: 67.472435, warp loss: 1.099877  [ 3480/    0]
> obs. loss: 0.237651, occl. loss:  0.142785, flow loss: 76.707634, warp loss: 1.101141  [ 3504/    0]
> obs. loss: 0.237271, occl. loss:  0.142269, flow loss: 86.882683, warp loss: 1.101937  [ 3528/    0]
> obs. loss: 0.236673, occl. loss:  0.141731, flow loss: 97.023819, warp loss: 1.100915  [ 3552/    0]
> ```

Found a similar phenomenon, did you solve this problem?

**Comment by hugocasa** [comment]
should be fixed by ce19122
